### PR TITLE
Datatype I/O fixes

### PR DESCRIPTION
**Pull request by mvdbeek**
trying a different newline replacement strategy and paramterizing unit test

### CODE REVIEW

**Comment by jmchilton** [comment]
This is amazing @mvdbeek - thanks for saving me from this, I was really getting lost in the weeds and everything here looks amazing now.

File: lib/galaxy/datatypes/sniff.py [hunk 1]
```diff
@@ -136,24 +151,20 @@ def convert_newlines(fname, in_place=True, tmp_dir=None, tmp_prefix="gxupload"):
...
     Transforms in place a 'sep' separated file to a tab separated one
-
-    >>> fname = get_test_fname('temp.txt')
-    >>> with open(fname, 'wt') as fh:
-    ...     _ = fh.write(u"1 2\\n3 4\\n")
-    >>> sep2tabs(fname)
-    (2, None)
-    >>> open(fname).read()
-    '1\\t2\\n3\\t4\\n'
     """
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/sniff.py]: Is sep2tabs actually used anywhere ? Doesn't look like it to me ...

**Comment by mvdbeek** [comment]
OK, this removes some dead code and should be much more resilient to files with non-standard encoding. Should fix https://github.com/galaxyproject/galaxy/issues/7412 and https://github.com/galaxyproject/galaxy/issues/7957

**Comment by jmchilton** [comment]
I'm going to backport all of this to 19.05, some of it (e.g. ed24664 probably should go back to 19.01 also).

**Pull request merged**